stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -354,6 +354,49 @@ function copyto!(A::AbstractSparseMatrixCSC, B::AbstractSparseMatrixCSC)
     return A
 end
 
+copyto!(A::AbstractMatrix, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+# Ambiguity resolution
+copyto!(A::PermutedDimsArray, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+
+function _sparse_copyto!(dest::AbstractMatrix, src::AbstractSparseMatrixCSC{T}) where {T}
+    dest === src && return dest
+    checkbounds(dest, axes(src)...)
+    fill!(dest, zero(T))
```

Comment:
`dest` does not seem to have the same eltype as `src`, and perhaps this is not necessary. But then you should have
```julia
fill!(dest, zero(eltype(dest)))
```
here.

Reply:
The idea here is that the not-stored zeros of the sparse matrix `src` have to be copied to `dest` before copying all the values that *are* stored. By using `fill!(dest, zero(T))` and relying on `convert` methods we do just that. Note that the line `dest[row, col] = val` further down also relies on conversion from `T` to `eltype(dest)`. This is analogous. If no conversion is possible between the two types we want this to throw an error, which it would not do if we just go for `fill!(dest, zero(eltype(dest)))`. Imagine the corner case of copying a sparse matrix with no stored values to a dense matrix with an incompatible `eltype`: if we do `fill!(dest, zero(eltype(dest)))` no error would be thrown.

Reply:
Then again, the same argument should be applied to the new `copyto!` method below, which does not rely on conversion for unstored entries. I will correct that.